lib/sink.js

```diff
@@ -47,38 +55,69 @@ class WriteStream extends stream.PassThrough {
     }
 }
 
-
 class ReadStream extends fs.createReadStream {
-    constructor (...args) {
+    constructor(...args) {
         super(...args);
         const file = path.parse(args[0]).base;
         this.on('open', () => {
             this.emit('file found', file);
         });
-        this.on('error', (error) => {
+        this.on('error', error => {
             if (error.code === 'ENOENT') {
                 this.emit('file not found', file);
             }
         });
     }
 }
 
-
 module.exports = class SinkFs extends EventEmitter {
-    constructor (options = {}) {
+    constructor(options = {}) {
         super();
         assert(options.path, '"options.path" is missing');
         this.name = 'asset-pipe-sink-fs';
         this.options = Object.assign({}, options);
     }
 
-    writer (type) {
+    getFileName(fileName) {
+        return path.join(this.options.path, fileName);
+    }
+
+    async get(fileName) {
+        assert(fileName, '"fileName" is missing');
+        try {
+            const file = await readFile(this.getFileName(fileName), 'utf8');
+            if (file) {
+                return file;
+            }
+        } catch (e) {}
+        throw new Error(`No file with fileName "${fileName}"`);
+    }
+
+    async set(fileName, fileContent) {
+        assert(fileName, '"filename" is missing');
+        assert(fileContent, '"fileContent" is missing');
+        await writeFile(this.getFileName(fileName), fileContent, 'utf8');
+    }
+
+    async has(fileName) {
+        assert(fileName, '"fileName" is missing');
+        try {
+            await access(
+                this.getFileName(fileName),
+                fs.constants.R_OK | fs.constants.W_OK
+            );
+            return true;
+        } catch (e) {}
+        return false;
+    }
+
+    writer(type) {
         assert(type, '"type" is missing');
         return new WriteStream(this.options, type);
     }
 
-    reader (file) {
-        assert(file, '"file" is missing');
-        return new ReadStream(path.join(this.options.path, file));
+    reader(fileName) {
+        assert(fileName, '"fileName" is missing');
```

Comment:
you could do something like: 
```js
assert(typeof fileName === 'string', `Expected "fileName" to be a string. Instead got ${fileName}`)
```
to catch more cases.

Reply:
https://github.com/asset-pipe/asset-pipe-sink-fs/commit/7514ea54d60f6c0522d57e2720ffee2bbace7a97